### PR TITLE
[tests] add test for Advertising Proxy to publish zero or multiple host addrs

### PR DESCRIPTION
**Pull request by superwhd**
Depends-On: openthread/ot-br-posix#1491

### CODE REVIEW

**Comment by wgtdkp** [comment]
> add a test case that Advertising Proxy publishes zero or multiple host addresses

@superwhd Are there test cases for publishing zero host addresses?

**Comment by superwhd** [comment]
> > add a test case that Advertising Proxy publishes zero or multiple host addresses
> 
> @superwhd Are there test cases for publishing zero host addresses?

Yes. See `tests/scripts/thread-cert/border_router/test_advertising_proxy.py:244`.

File: tests/scripts/thread-cert/node.py [hunk 1]
```diff
@@ -3417,7 +3419,7 @@ def discover_mdns_service(self, instance, name, host_name, timeout=2):
                 assert (service['host_fullname'] == f'{host_name}.local.')
                 service['host'] = host_name
                 service['addresses'] = addresses
-        return service if 'addresses' in service and service['addresses'] else None
+        return service if service else None
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/node.py:3422]:
```suggestion
        return service or None
```
very nit

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

File: tests/scripts/thread-cert/border_router/test_advertising_proxy.py [hunk 2]
```diff
@@ -230,10 +266,11 @@ def host_check_mdns_service(self, host, host_addr, service_instance, service_por
         self.assertEqual(service['priority'], 0)
         self.assertEqual(service['weight'], 0)
         self.assertEqual(service['host'], 'my-host')
-        self.assertEqual(ipaddress.ip_address(service['addresses'][0]), ipaddress.ip_address(host_addr))
-        self.assertEqual(len(service['addresses']), 1)
+        self.assertEqual(len(service['addresses']), len(host_addrs))
+        for addr1, addr2 in zip(sorted(service['addresses']), sorted(host_addrs)):
+            self.assertEqual(ipaddress.ip_address(addr1), ipaddress.ip_address(addr2))
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_advertising_proxy.py:271]:
```suggestion
        self.assertEqual(sorted(service['addresses']), sorted(host_addrs))
```
nit

File: tests/scripts/thread-cert/border_router/test_advertising_proxy.py [hunk 3]
```diff
@@ -208,20 +208,56 @@ def test(self):
...
+        # 10. Check if the expired service is removed by the Advertising Proxy.
+        #
         client.srp_client_stop()
         self.simulator.go(LEASE + 2)
 
         self.assertIsNone(host.discover_mdns_service('my-service', '_ipps._tcp', 'my-host'))
         self.assertIsNone(host.discover_mdns_service('my-service-1', '_ipps._tcp', 'my-host'))
 
-    def host_check_mdns_service(self, host, host_addr, service_instance, service_port=12345):
+    def host_check_mdns_service(self, host, host_addrs, service_instance, service_port=12345):
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_advertising_proxy.py:260]:
```suggestion
    def host_check_mdns_service(self, host, host_addrs:Union[str, List[str]], service_instance, service_port=12345):
```
allowing both `str` for one addr or list of addrs may reduce code change.
But it's nit.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

with nits

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]
